Patch from upstream to fix segfault introduced in 2.46.0 that
causes Nautilus to crash on startup in our GNOME 2.30 desktop.

From 30359e740953b596d6c3d980dba36dd131681a1f Mon Sep 17 00:00:00 2001
From: Allison Ryan Lortie <desrt@desrt.ca>
Date: Mon, 30 Nov 2015 10:13:46 -0500
Subject: [PATCH] file monitors: reorder some code to avoid segfault

We must initialise '->source' before we use fields inside of it.

https://bugzilla.gnome.org/show_bug.cgi?id=758823
---
 gio/glocalfilemonitor.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gio/glocalfilemonitor.c b/gio/glocalfilemonitor.c
index dbd62ec..49a4dd2 100644
--- a/gio/glocalfilemonitor.c
+++ b/gio/glocalfilemonitor.c
@@ -748,6 +748,9 @@ g_local_file_monitor_start (GLocalFileMonitor *local_monitor,
 
   g_assert (!local_monitor->source);
 
+  source = g_file_monitor_source_new (local_monitor, filename, is_directory, flags);
+  local_monitor->source = source; /* owns the ref */
+
   if (is_directory && !class->mount_notify && (flags & G_FILE_MONITOR_WATCH_MOUNTS))
     {
 #ifdef G_OS_WIN32
@@ -771,9 +774,6 @@ g_local_file_monitor_start (GLocalFileMonitor *local_monitor,
 #endif
     }
 
-  source = g_file_monitor_source_new (local_monitor, filename, is_directory, flags);
-  local_monitor->source = source; /* owns the ref */
-
   G_LOCAL_FILE_MONITOR_GET_CLASS (local_monitor)->start (local_monitor,
                                                          source->dirname, source->basename, source->filename,
                                                          source);
-- 
2.7.4

